src/lib/components/toast/Toast.component.tsx

```diff
@@ -0,0 +1,180 @@
+import { motion } from 'framer-motion';
+import React, { ReactNode, useRef } from 'react';
+import { useTheme } from 'styled-components';
+import { Box } from '../box/Box';
+import { Button } from '../buttonv2/Buttonv2.component';
+import { Icon, IconColor } from '../icon/Icon.component';
+import { BasicText } from '../text/Text.component';
+import { ToastPosition, positionOutput } from './ToastPositionHelpers';
+import { ToastProgressBar } from './ToastProgressBar';
+import { useToastParameters } from './useToastParameters';
+import styled from 'styled-components';
+
+export type ToastStatus = 'success' | 'error' | 'warning' | 'info';
+
+export type ToastProps = {
+  open: boolean;
+  message: ReactNode;
+  onClose: () => void;
+  status?: ToastStatus;
+  position?: ToastPosition;
+  autoDismiss?: boolean;
+  duration?: number;
+  icon?: React.ReactNode;
+  width?: React.CSSProperties['width'];
+  withProgressBar?: boolean;
+  style?: React.CSSProperties;
+};
+
+export const useGetBackgroundColor = (status: string) => {
+  const theme = useTheme();
+  switch (status) {
+    case 'success':
+      return theme.statusHealthy;
+    case 'error':
+      return theme.statusCritical;
+    case 'warning':
+      return theme.statusWarning;
+    default:
+      return theme.infoPrimary;
+  }
+};
+
+const useGetRgbBackgroundColor = (status: string) => {
+  const theme = useTheme();
+  switch (status) {
+    case 'success':
+      return 'rgba(10, 173, 166, 0.4)';
+    case 'error':
+      return 'rgba(232, 72, 85, 0.4)';
+    case 'warning':
+      return 'rgba(248, 243, 43, 0.4)';
+    default:
+      return theme.infoSecondary;
+  }
+};
+
+const defaultIconName = (status: string) => {
+  switch (status) {
+    case 'success':
+      return 'Check-circle';
+    case 'error':
+      return 'Times-circle';
+    case 'warning':
+      return 'Exclamation-circle';
+    default:
+      return 'Info-circle';
+  }
+};
+
+const DefaultIcon = ({ status }: { status: string }) => {
+  const color = useGetBackgroundColor(status);
+  const iconName = defaultIconName(status);
+
+  return <Icon name={iconName} color={color as IconColor} />;
```

Comment:
Why do we need to cast the IconColor type here ? Is this type not accurate ? maybe we can update it if so 🤔